### PR TITLE
Refact deploy script to support gpg and nvsec signature [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
CI got PASS for both `gpg` & `nvsec` signature

File: jenkins/deploy.sh [hunk 1]
```diff
@@ -17,76 +17,81 @@
...
 if [ "$SIGN_FILE" == true ]; then
-    # No javadoc and sources jar is generated for shade artifact only. Use 'sql-plugin' instead
-    SQL_ART_ID=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.artifactId -DforceStdout`
-    SQL_ART_VER=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.version -DforceStdout`
-    JS_FPATH="${SQL_PL}/target/spark${FINAL_AGG_VERSION_TOBUILD}/${SQL_ART_ID}-${SQL_ART_VER}"
-    SRC_DOC_JARS="-Dsources=${JS_FPATH}-sources.jar -Djavadoc=${JS_FPATH}-javadoc.jar"
-    DEPLOY_CMD="$MVN -B gpg:sign-and-deploy-file -s jenkins/settings.xml -Dgpg.passphrase=$GPG_PASSPHRASE"
+    case $SIGN_TOOL in
+        nvsec)
+            cp $NVSEC_CFG_FILE ~/.nvsec.cfg
```
Comment (jlowe) [inline jenkins/deploy.sh:69]: This is a bit surprising.  What if the user running this script had their own custom settings in ~/.nvsec.cfg?  Are we sure there's no way to pass an explicit config path to nvsec?

Comment (NvTimLiu) [reply jenkins/deploy.sh:69]: What if the user running this script had their own custom settings in ~/.nvsec.cfg? 
     --> The file `.nvsec.cfg` is only for signing jars with the `nvsec` tool, the additional Sonatype credentials to are needed to deploy to OSS staging repo. External user should have no secrets of `.nvsec.cfg`, so they can not sign via `nvsec`

Are we sure there's no way to pass an explicit config path to nvsec? 
    --> There is a way to pass config path to nvsec, but we are wrapping `nvsec` as the `gpg` executable of the plugin `gpg:sign-and-deploy-file`, so we need to PASS the `.nvsec.cfg` path via gpg plugin. This is a workaround and odd way, as GPG plugin need not such file.

Let me discuss with @pxLi  later to see if we have better way te set nvsec credentials, maybe apply nvsec secret from Jenkinsfile is a way.

Comment (pxLi) [reply jenkins/deploy.sh:69]: > maybe apply nvsec secret from Jenkinsfile is a way.
yeah, I agree to put the nvsec config setup in pipeline side

Comment (NvTimLiu) [reply jenkins/deploy.sh]: fixed, moved nvsec config setup into pipeline side

**Comment by NvTimLiu** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by pxLi — APPROVED** [review]
Please also help update JNI deploy script later, thx~

**Pull request merged**